homeassistant/components/unifi/sensor.py

```diff
@@ -218,16 +219,29 @@ def async_client_wan_monitor_supported_fn(
 
 @callback
 def async_client_wan_monitor_latency(
-    wan: TypedDeviceUptimeStatsWan | None, monitor_target: str
+    wan: Literal["WAN", "WAN2"],
+    monitor_target: str,
+    hub: UnifiHub,
+    device: Device,
 ) -> int | None:
     """Retrieve the monitor target from WAN monitors."""
 
-    if wan is None:
-        return None
+    uptime_stats = device.uptime_stats
 
-    monitors = wan["monitors"]
+    if uptime_stats is None:
+        return False
+
+    uptime_stats_wan = uptime_stats[wan]
```

Comment:
For future safety, it would be good to make sure the key is present.

You could do:

```python
if uptime_stats_wan := uptime_stats.get(wan):
    for item in uptime_stats_wan["monitors"]:
        if monitor_target in item["target"]:
            return item.get("latency_average", 0)

return None
```